test/implementations/lazyMintNativeTokenPay.ts

```diff
@@ -145,6 +150,21 @@ async function shouldControlValidMintingNativeTokenPay(): Promise<void> {
     expect(await this.token.balanceOf(addrs[0].address)).to.equal(1);
   });
 
+  it('reduces total supply on burn', async function () {
+    const tokenId = await mintFromImplNativeToken(this.token, addrs[0].address);
+    await this.token.connect(addrs[0])['burn(uint256)'](tokenId);
+    expect(await this.token.totalSupply()).to.equal(0);
```

Comment:
I'd also validate that the `totalSupply` before the burn is `1`

Reply:
Or at least that it si greater before the burn than after the burn.

Reply:
Good call, added it.